octodns/provider/azuredns.py

```diff
@@ -613,8 +626,13 @@ def _data_for_CNAME(self, azrecord):
 
             :type  return: dict
         '''
-        if azrecord.cname_record is None and azrecord.target_resource.id:
-            return self._data_for_dynamic(azrecord)
+        if azrecord.cname_record is None:
+            if azrecord.target_resource.id:
+                return self._data_for_dynamic(azrecord)
+            else:
+                # dynamic record alias is broken, return dummy value and apply
+                # will likely overwrite/fix it
```

Comment:
Probably worth a warning log here with a bit more info about what's happening, like the comment.

Reply:
pushed new commit for this